src/primitives/arith.rs

```diff
@@ -962,7 +962,7 @@ pub fn deep_quotient_except_public_inputs<'a, 'b>(
     challenges: &[EF],
     denom_at_z: &ComplexPoly<'a, CosetEvaluations>,
     denom_at_z_omega: &ComplexPoly<'a, CosetEvaluations>,
-    denom_at_zero: &Poly<'a, CosetEvaluations>,
+    maybe_denom_at_zero: &Option<Poly<'a, CosetEvaluations>>,
```

Comment:
Usually `Option` should be passed by value, in this case that would be option to reference instead of reference to option: `Option<&Poly<'a, CosetEvaluations>>`. This would also save you the `as_ref()` calls.
Also consider changing the other "reference to option to reference" parameters like `maybe_witness_cols: &Option<&Vec<Poly<'a, CosetEvaluations>>>` to just `maybe_witness_cols: Option<&Vec<Poly<'a, CosetEvaluations>>>`

Reply:
Let's leave it as is for now, can refactor later.